pkg/cloudprovider/interface.go

```diff
@@ -24,6 +25,9 @@ type CloudProvider interface {
 	// Refresh is called before every main loop and can be used to dynamically update cloud provider state.
 	// In particular the list of node groups returned by NodeGroups can change as a result of CloudProvider.Refresh().
 	Refresh() error
+
+	// NodeInstantiationTime gets the time that the cloud provider created the resource used to back the node
+	NodeInstantiationTime(node *v1.Node) time.Time
```

Comment:
I think NodeInstantiationTime should really be the following:
```go
NodeInfo(node *v1.Node) (Node, error)
```
The Node object that it returns is a cloudprovider node interface like so:
```go
type Node interface {
	...
	InstantiationTime() time.Time
	...
}
```
This way we can have a nice way to describe nodes in the cloud provider and extract more info from the cloud provider.

NodeInfo will also need to be tested with the MockAWS cloud provider too.